### PR TITLE
Fix: Rename some protocol classes

### PR DESCRIPTION
**Pull request by ueman**
## :scroll: Description

This is a _breaking change_.

This PR renames a couple classes: 
- App -> SentryApp
- Browser -> SentryBrowser
- Device -> SentryDevice
- Message -> SentryMessage
- Request -> SentryRequest
- User -> SentryUser
- Gpu -> SentryGpu
- OperatingSystem -> SentryOperatingSystem
- Integration -> SentryIntegration

I wonder if we should rename the following classes too:
- Breadcrumb
- Contexts

The following classes are probably fine:
- DebugImage
- DebugMeta
- Dsn
- Mechanism
- SdkInfo
- SdkVersion 

## :bulb: Motivation and Context
See #250 

## :green_heart: How did you test it?
No functional changes, tests are still green

## :pencil: Checklist
- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated the docs if needed
- [x] All tests passing
- [ ] No breaking changes

## :crystal_ball: Next steps
Update documentation:
- https://github.com/getsentry/sentry-docs/pull/3244

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 * Fix: Multiple FlutterError.onError calls in FlutterErrorIntegration (#345)
 * Fix: Pass hint to EventProcessors (#356)
 * Fix: EventProcessors were not dropping events when returning null (#353)
+* Fix: Prefix App, Browser, Device, Message, Request and User with Sentry (#250)
```
Comment (marandaneto) [inline CHANGELOG.md]:
```suggestion
* Fix: Prefix App, Browser, Device, Message, Request and User with Sentry (#357)
```

Comment (marandaneto) [reply CHANGELOG.md]: we're used to link the PR's id, or should we link the issue's id? wondering which one does make more sense, this suggestion is just to discuss, the Danger CI checks the PRs id btw

Comment (marandaneto) [inline CHANGELOG.md]:
```suggestion

Breaking Changes:

* Fix: Prefix App, Browser, Device, Message, Request and User with Sentry (#250)

// let's describe here the renamed Protocols and that now they have a 'Sentry' prefix.
```

Comment (marandaneto) [reply CHANGELOG.md]: we'll probably need to add a Migration page to https://docs.sentry.io/platforms/dart/ but let's track this in another issue :)

**Comment by marandaneto** [comment]
I guess `Breadcrumb` and `Contexts` are fine.
`OperatingSystem`, `Integration` and `Gpu` I'd have renamed though, now with more supported platforms, that might become a problem too.
@bruno-garcia @philipphofmann ?

File: CHANGELOG.md (hunk 1 above)
Comment (ueman) [reply CHANGELOG.md]: Ah I see, I already wondered why it failed. 

I think both solutions are fine because PR and issue get linked. So lets keep PR IDs 👍

**Review by bruno-garcia — APPROVED** [review]
LGTM but I wonder if we're going into 5.0.0 now?

**Comment by bruno-garcia** [comment]
> `OperatingSystem`, `Integration` and `Gpu` I'd have renamed though, now with more supported platforms, that might become a problem too.
> @bruno-garcia @philipphofmann ?

Agree renaming these 3 could make sense.

**Comment by marandaneto** [comment]
> LGTM but I wonder if we're going into 5.0.0 now?

let's do one more pre-release with the fixes we've done and if all good, we do a GA next.

**Comment by marandaneto** [comment]
> > `OperatingSystem`, `Integration` and `Gpu` I'd have renamed though, now with more supported platforms, that might become a problem too.
> > @bruno-garcia @philipphofmann ?
> 
> Agree renaming these 3 could make sense.

@ueman do you prefer to rename these 3 left on this PR or a new one?

**Comment by ueman** [comment]
> > > `OperatingSystem`, `Integration` and `Gpu` I'd have renamed though, now with more supported platforms, that might become a problem too.
> > > @bruno-garcia @philipphofmann ?
> > 
> > 
> > Agree renaming these 3 could make sense.
> 
> @ueman do you prefer to rename these 3 left on this PR or a new one?

I've renamed them in this PR.

**Comment by philipphofmann** [comment]
For consistency, we could prefix all of them with Sentry. As a user, I can imagine, that you don't fully understand why some of them have the prefix and some of them don't.

**Pull request merged**